File: algorithm/그래프/BFS_DFS/DFS2.py
graph = [
    [],
    [2, 3, 8],
    [1, 7],
    [1, 4, 5],
    [3, 5],
    [3, 4],
    [7],
    [2, 6, 8],
    [1, 7]
]

visited = [False] * 9


def dfs(graph, v, visited):
    visited[v] = True

    for i in graph[v]:  # 2,3,8
        if not visited[i]:  # 2,3,8 true 라면 다시
            dfs(graph, i, visited)  # 그래프와 2,3,8 순회한다


dfs(graph, 1, visited)

File: algorithm/그래프/BFS_DFS/dfs2.js
let graph2 = [
  [],
  [2, 3, 8],
  [1, 7],
  [1, 4, 5],
  [3, 5],
  [3, 4],
  [7],
  [2, 6, 8],
  [1, 7],
];
let visited = Array(9).fill(false);

function dfs2(graph, v, visited) {
  visited[v] = true;
  console.log(v);
  for (let i of graph[v]) {
    if (!visited[i]) {
      dfs2(graph, i, visited);
    }
  }
}

console.log(dfs2(graph2, 1, visited));
